swift/common/shardtrie.py

```diff
@@ -142,8 +142,8 @@ def popping_iter(self):
         yield self
         if isinstance(self.parent, Node):
             del self.parent.children[self.key]
-        for child in sorted(self.children):
-            for c in child.popping_iter():
+        for child in sorted(self.children.items(), lambda x, y: x[0] < y[0]):
+            for c in child[1].popping_iter():
                 yield c
```

Comment:
We only care about the sorting of the keys, so does it look cleaner (and similar in other accessing children cases in the class) to: 

```
  for child in sorted(self.children):
      for c in self.children[child].popping_iter():
           yield c
```
Although I don't feel too strongly about either approach. So this is probably more of a nitpic :P